src/diffusers/pipelines/stable_diffusion/convert_from_ckpt.py

```diff
@@ -1007,13 +1016,92 @@ def load_pipeline_from_original_stable_diffusion_ckpt(
+            Pass `None` to determine automatically based on checkpoint file extension
+        pipeline_class (`str`, *optional*, defaults to `None`):
+            The pipeline class to use. Pass `None` to determine automatically.
+
+        
     """
+
+    # import pipelines here to avoid circular import error when using from_ckpt method
+    from diffusers import (
+        StableDiffusionControlNetPipeline,
+        StableDiffusionPipeline,
+        StableUnCLIPImg2ImgPipeline,
+        StableUnCLIPPipeline,
+        LDMTextToImagePipeline,
+        PaintByExamplePipeline
+    )
+
+    if pipeline_class:
+
+        pipeline_name = pipeline_class.__name__
+
+        if pipeline_name == "StableDiffusionControlNetPipeline":
+            model_type = "FrozenCLIPEmbedder"
+            controlnet = True
+
+        elif "StableDiffusion" in pipeline_name:
+            model_type = "FrozenCLIPEmbedder"
+            controlnet = False
+
+        elif pipeline_name == "StableUnCLIPImg2ImgPipeline":
+            model_type == "FrozenOpenCLIPEmbedder"
+            stable_unclip = "img2img"
+
+        elif pipeline_name == "StableUnCLIPPipeline":
+            model_type == "FrozenOpenCLIPEmbedder"
+            stable_unclip = "txt2img"
+
+        elif pipeline_name == "PaintByExamplePipeline":
+            model_type == "PaintByExample"
+
+        elif pipeline_name == "LDMTextToImagePipeline":
+            model_type == "LDMTextToImage"
+
+        else:
+            raise ValueError(f"Unhandled pipeline class: {pipeline_name}")
+
+    else:
+        pipeline_class = StableDiffusionPipeline
+
     if prediction_type == "v-prediction":
         prediction_type = "v_prediction"
 
+    if from_safetensors is None:
+        from_safetensors = checkpoint_path.rsplit(".", 1)[-1] == "safetensors"
+
+    # kwargs used for downloading both ckpt and config files
+    proxies = kwargs.pop("proxies", None)
+    use_auth_token = kwargs.pop("use_auth_token", None)
+
+    # Code based on diffusers.pipelines.pipeline_utils.DiffusionPipeline.from_pretrained
+    if not os.path.isfile(checkpoint_path):
+        checkpoint_path = Path(checkpoint_path)
+
+        # get repo_id and (potentially nested) file path of ckpt in repo
+        repo_id = str(Path().joinpath(*checkpoint_path.parts[:2]))
+        file_path = str(Path().joinpath(*checkpoint_path.parts[2:]))
+
+        cache_dir = kwargs.pop("cache_dir", DIFFUSERS_CACHE)
+        resume_download = kwargs.pop("resume_download", False)
+        local_files_only = kwargs.pop("local_files_only", HF_HUB_OFFLINE)
+        revision = kwargs.pop("revision", None)
+
+        checkpoint_path = hf_hub_download(
+            repo_id,
+            filename=file_path,
+            cache_dir=cache_dir,
+            resume_download=resume_download,
+            proxies=proxies,
+            local_files_only=local_files_only,
+            use_auth_token=use_auth_token,
+            revision=revision,
+        )
```

Comment:
we shouldn't need to make this many changes to the existing function. any changes for downloading from the hub can be self contained in the calling function